website/payments/models.py

```diff
@@ -32,19 +32,17 @@ class Meta:
     @classmethod
     def tpay_enabled(cls):
         today = timezone.now().date()
-        return Case(
-            When(
-                Q(
-                    bank_accounts__valid_from__isnull=False,
-                    bank_accounts__valid_from__lte=today,
-                )
-                & (
-                    Q(bank_accounts__valid_until__isnull=True)
-                    | Q(bank_accounts__valid_until__gt=today)
-                ),
-                then=settings.THALIA_PAY_ENABLED_PAYMENT_METHOD,
-            ),
-            default=False,
+        return Count(
+            "bank_accounts__pk",
+            filter=Q(
+                bank_accounts__valid_from__isnull=False,
+                bank_accounts__valid_from__lte=today,
+            )
+            & (
+                Q(bank_accounts__valid_until__isnull=True)
+                | Q(bank_accounts__valid_until__gt=today)
+            )
+            & Value(settings.THALIA_PAY_ENABLED_PAYMENT_METHOD),
```

Comment:
```suggestion
return Case(
            When(
                Exists(
                    BankAccount.objects.filter(owner=OuterRef("pk"))
                    .filter(valid_from__isnull=False, valid_from__lte=today)
                    .filter(valid_until__isnull=True, valid_until__gt=today)
                ),
                then=settings.THALIA_PAY_ENABLED_PAYMENT_METHOD,
            ),
            default=False,
```

Reply:
almost, there's missing a disjunction in your filter criteria

Reply:
ah good catch